### PR TITLE
Master production line whe

### PR DESCRIPTION
**Pull request by Whenrow**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_round, float_is_zero
+
+
+class MrpAbstractWorkorder(models.AbstractModel):
+    _name = "mrp.abstract.workorder"
+    _description = "basic features and fields shared between production wizard\
+    and workorders"
+
+    production_id = fields.Many2one('mrp.production', 'Manufacturing Order')
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: required?

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    product_id = fields.Many2one(related='production_id.product_id', readonly=True, store=True)
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: required?

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+from odoo import api, fields, models, _
+from odoo.addons import decimal_precision as dp
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_round, float_is_zero
+
+
+class MrpAbstractWorkorder(models.AbstractModel):
+    _name = "mrp.abstract.workorder"
+    _description = "basic features and fields shared between production wizard\
+    and workorders"
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: make it one line

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+
+    production_id = fields.Many2one('mrp.production', 'Manufacturing Order')
+    product_id = fields.Many2one(related='production_id.product_id', readonly=True, store=True)
+    qty_producing = fields.Float(string='Currently Produced Quantity', digits=dp.get_precision('Product Unit of Measure'))
+    product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', readonly=True)
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: maybe put the onchange with the domain of the uom's category in the abstract model?

File: addons/mrp/models/mrp_abstract_workorder.py (hunk 3 above)
Comment (sle-odoo) [reply addons/mrp/models/mrp_abstract_workorder.py]: Common code between produce wizards and workorders.

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+        return vals_list
+
+    def _find_candidate(self, move_line):
+        """ Method used in order to return move lines that match reservation.
+        The purpose is to update exisiting workorder line regarding the
+        reservation of the raw moves.
+        """
+        res = self.filtered(lambda line:
+            line.lot_id == move_line.lot_id and
+            line.qty_done < move_line.product_uom_qty and
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: float_compare

File: addons/mrp/models/mrp_abstract_workorder.py [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+            vals_list.append(vals)
+
+        return vals_list
+
+    def _find_candidate(self, move_line):
+        """ Method used in order to return move lines that match reservation.
+        The purpose is to update exisiting workorder line regarding the
+        reservation of the raw moves.
+        """
+        res = self.filtered(lambda line:
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_abstract_workorder.py]: you could return directly

File: addons/mrp/models/mrp_abstract_workorder.py (hunk 2 above)
Comment (amoyaux) [reply addons/mrp/models/mrp_abstract_workorder.py]: related

**Comment by sle-odoo** [comment]
robodoo rebase-ff r+

**Comment by sle-odoo** [comment]
robodoo rebase-ff r+

**Comment by Whenrow** [comment]
retry

**Comment by Whenrow** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**